### PR TITLE
Add git version hash to sources and log file

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #87

This logs the git commit hash in the output log. Should help with bug reports.

For cmake this uses some cmakefiles from here: https://github.com/rpavlik/cmake-modules/blob/master/GetGitRevisionDescription.cmake

These will correctly rebuild the `git_version.cpp` each time the git hash changes.

Plaforms:
- [x] cmake
- [x] autotools
- [x] Android
- [x] 3ds
- [ ] switch
- [x] gcw0
- [x] wii
- [x] vita

Features
- [x] Detects git version correctly
- [x] Reacts to git HEAD changes and rebuilds
- [x] Builds from source tarball without `.git` - `make distcheck` in autotools

Todo:
- [ ] Remove PLAYER_ADDTL_VERSION hacks from jenkins build scripts (not part of the codebase)

### CODE REVIEW

File: builds/git_version.mk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+GIT_ROOT = $(REPO_ROOT)/.git
+GIT_HEAD = $(GIT_ROOT)/HEAD
+GIT_REV_FILE = $(shell $(REPO_ROOT)/builds/git_ref.sh $(GIT_HEAD))
+
```
Comment (carstene1ns) [inline builds/git_version.mk]: Can you explain why all of this magic is needed? [`git rev-parse`](https://git-scm.com/docs/git-rev-parse) will do exactly what is needed here.



fine tuned: `git rev-parse --verify -q --short=10 HEAD^{commit}`

Comment (fmatthew5876) [reply builds/git_version.mk]: We can and probably should use git to generate the string. However we need to figure out which file contains the hash and add it as a dependency of the rule. This ensures that `git_version.cpp` is regenerated whenever `HEAD` changes.

Comment (carstene1ns) [reply builds/git_version.mk]: `git rev-parse --absolute-git-dir`
`git symbolic-ref -q HEAD`
?

**Comment by carstene1ns** [comment]
Closing in favor of #2774.

**Pull request closed without merging**